data-generator/src/main/java/cloud/project/datagenerator/hotels/HotelsDataGenerator.java

```diff
@@ -0,0 +1,132 @@
+import cloud.project.datagenerator.hotels.domain.Room;
+import cloud.project.datagenerator.hotels.repositories.HotelRepository;
+import cloud.project.datagenerator.rabbitmq.QueuesConfig;
+import cloud.project.datagenerator.rabbitmq.json.JsonConverter;
+import cloud.project.datagenerator.rabbitmq.requests.RoomUpdateRequest;
+import lombok.RequiredArgsConstructor;
+import org.springframework.amqp.rabbit.core.RabbitTemplate;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+import java.util.Random;
+import java.util.UUID;
+
+@Component
+@RequiredArgsConstructor
+public class HotelsDataGenerator {
+    enum DataUpdateType {
+        CREATE,
+        UPDATE,
+        DELETE
+    }
+
+    private final HotelRepository hotelRepository;
+    private final Random random = new Random();
+    private final RabbitTemplate rabbitTemplate;
+
+    @Scheduled(fixedDelay = 5000, initialDelay = 5000)
+    public void updateRandomHotelData() {
+        int action = random.nextInt(3);
+
+        switch (action) {
+            case 0:
+                createNewRoom();
+                break;
+            case 1:
+                updateRandomRoom();
+                break;
+            case 2:
+                deleteRandomRoom();
+                break;
+        }
+    }
+
+    private void createNewRoom() {
+        Hotel randomHotel = getRandomHotel();
+        if (randomHotel == null) return;
+
+        Room newRoom = Room.builder()
+                .id(UUID.randomUUID())
+                .hotel(randomHotel)
+                .name("NewRoom-" + random.nextInt(100))
+                .guestCapacity(random.nextInt(3, 10))
+                .pricePerAdult(random.nextFloat() * 100)
+                .description("This is an awesome new room generated by Data Generator module")
+                .build();
+
+        updateHotelDataInHotelModules(DataUpdateType.CREATE, newRoom);
+    }
+
+    private void updateRandomRoom() {
+        Hotel randomHotel = getRandomHotel();
+        if (randomHotel == null) return;
+
+        Room randomRoom = randomHotel.getRooms().get(random.nextInt(randomHotel.getRooms().size()));
+        int currentGuestCapacity = randomRoom.getGuestCapacity();
+        int newGuestCapacity = random.nextInt(1, currentGuestCapacity + 10);
+
+        float currentPricePerAdult = randomRoom.getPricePerAdult();
+        float newPricePerAdult = random.nextFloat(currentPricePerAdult, currentPricePerAdult*10);
+
+        randomRoom.setGuestCapacity(newGuestCapacity);
+        randomRoom.setPricePerAdult(newPricePerAdult);
+
+        updateHotelDataInHotelModules(DataUpdateType.UPDATE, randomRoom);
+    }
+
+    private void deleteRandomRoom() {
+        Hotel randomHotel = getRandomHotel();
+        if (randomHotel == null) return;
+
+        Room randomRoom = randomHotel.getRooms().get(random.nextInt(randomHotel.getRooms().size()));
```

Comment:
to raczej nie przejdzie, musi być sprawdzenie czy dany pokój nie został zarezerwowany/kupiony. Tyczy się to także transportów

Reply:
żeby to usuwanie zrobić to wgl trzeba by było tu jakiś potężny system chyba strzelić żeby to hotele/transporty zwracały które elementy nie są zajęte, ale to dużo pracy więc póki co najlepiej chyba byłoby to robić bez usuwania

Reply:
Okej, to usunę te metody z usuwaniem pokoi i transportów, bo rzeczywiście sporo pracy to może kosztować, a bez tego wymagania i tak są spełnione

Reply:
Usunięte